src/lib/Scenes/Map/Components/PinsShapeLayer.tsx

```diff
@@ -52,7 +58,7 @@ export class ShapeLayer extends Component<Props, any> {
   }
 
   componentWillReceiveProps(newProps: Props) {
-    if (!isEqual(this.props.featureCollection.features, newProps.featureCollection.features)) {
+    if (!isEqual(this.props.featureCollections[0].shapes.features, newProps.featureCollections[0].shapes.features)) {
```

Comment:
I'm not sure about  `featureCollections` being an array, I would say an object where the key is the filter type makes more sense. Lookup times are faster